client/purplecaffeine/trial.py

```diff
@@ -0,0 +1,129 @@
+"""Trial."""
+import os
+from typing import Optional, Union, List, Any
+import numpy as np
+from qiskit.providers import Backend
+from qiskit.circuit import QuantumCircuit
+from qiskit.quantum_info.operators import Operator
+
+from purplecaffeine.helpers import Configuration
+from purplecaffeine.backend import BaseBackend, LocalBackend
+
+
+class Trial:
+    """Trial class."""
+
+    def __init__(self, name: str, backend: Optional[BaseBackend] = None):
+        """Trial class for tracking experiments data.
+
+        Args:
+            name: name of trial
+            backend: backend to store data of trial. Default: local storage.
+
+        Properties:
+            metrics: list of metric, like number of qubits
+            parameters: list of parameter, like env details
+            circuits: list of quantum circuit
+            qbackends: list of quantum backend
+            operators: list of operator, like Pauli operators
+            artifacts: list of artifact path, any external files path
+            texts: list of text, any descriptions
+            arrays: list of array, like quantum circuit results
```

Comment:
I didn't see any specific way to indicates them in the DocString doc, so i guess like this could pass ^^